test/distributed/_composable/test_compose.py

```diff
@@ -208,6 +209,30 @@ def test_checkpoint_fsdp_submodules_with_param(self):
                 self._test_parity,
             )
 
+    @skip_if_lt_x_gpu(2)
+    def test_checkpoint_fsdp_submodules_with_param_no_shard(self):
+        model = CompositeParamModel().to(torch.device("cuda"))
+
+        base_model = copy.deepcopy(model)
+
+        test_model = copy.deepcopy(model)
+        test_model.u1.seq = checkpoint(test_model.u1.seq, use_reentrant=False)
+        test_model.u2.seq = checkpoint(test_model.u2.seq, use_reentrant=False)
+        test_model = fully_shard(test_model, strategy=ShardingStrategy.NO_SHARD)
+
+        with self.assertRaisesRegex(
+            RuntimeError, "Cannot writeback when the parameter shape changes"
```

Comment:
Is this repro-like test supposed to be part of this PR or just meant for debugging?

Reply:
I was intended to land this test. When the error is fixed, we can just remove this `assertRaisesRegex` so that future changes won't break this use case.